### DISCUSSION TITLE
Plotly express question

### DISCUSSION THREAD

**Discussion by bufordbufordson** [pycaret/pycaret#4048] [Q&A]
I have this code. I’m trying to create a map with the plotly_express markers showing a certain text (from the type column, which is included) right on top of the markers, hence the text = 'Type' and textposition = 'top center'. (See code below.) But when I run the code, the text only appears when I hover, which is the opposite of what I’m looking for. How to fix? Is it because I am missing a mapbox token?

fig = px.scatter_mapbox(
   daily_average_exceedances_df,
   lat = 'Latitude',
   lon = 'Longitude',
   color='Daily Exceedances',
   color_continuous_scale=px.colors.sequential.Sunset,
   range_color=(0,30),
   hover_data=['siteId'], 
   text=daily_average_exceedances_df['Type'], 
).update_layout(
   mapbox={"style": "carto-positron", "zoom":11}, margin={"l": 0, "r": 0, "t": 0, "b": 0}
)

Thanks,
Buford Bufordson

**Comment by Deadcoder001** [comment]
Try this:
fig = go.Figure(go.Scattermapbox(
    lat=daily_average_exceedances_df['Latitude'],
    lon=daily_average_exceedances_df['Longitude'],
    mode='markers+text',
    marker=go.scattermapbox.Marker(
        size=9,
        color=daily_average_exceedances_df['Daily Exceedances'],
        colorscale='Sunset',
        cmin=0,
        cmax=30,
        colorbar=dict(title="Daily Exceedances"),
    ),
    text=daily_average_exceedances_df['Type'],
    textposition='top center',
    hoverinfo='text+lon+lat'
))
fig.update_layout(
    mapbox=dict(
        style="carto-positron",
        zoom=11,
        center=dict(lat=daily_average_exceedances_df['Latitude'].mean(), lon=daily_average_exceedances_df['Longitude'].mean())
    ),
    margin={"l": 0, "r": 0, "t": 0, "b": 0}
)
fig.show()

**Comment by celestinoxp** [comment]
@Deadcoder001 matplotlib is set to version lower than 3.8.0. Would you be interested in contributing so that pycaret can use the latest version of matplotlib? It would be interesting if you could submit a pull request

**Reply by Deadcoder001** [reply]
i dont know bro maybe i can try.

**Reply by bufordbufordson** [reply]
Thank you. I still don't see the text unless I hover over each marker. Maybe it's the mapbox access token that's missing?
